# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/joj48/Flyrank.ai-SEO-Project/blob/main/work/notebooks/w01_research_question.ipynb)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

## My Lane: Refresh / Content Opportunity Scoring

I selected **Refresh / Content Opportunity Scoring** because the starter dataset and pipeline are designed to prioritize webpages that should be reviewed for content updates. The dataset contains search performance, engagement, freshness, and trend-related metrics that can help identify pages with the highest potential for improvement. This lane focuses on supporting SEO and content teams by ranking refresh opportunities instead of treating every page equally.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

### Research Question

How can historical search performance and content freshness metrics be used to identify webpages that should be prioritized for content refresh?

**Decision**

Determine which webpages should be reviewed and refreshed first.

**Who acts on it?**

SEO specialists and content editors who maintain website content.

**Output**

A ranked list of webpages with a content opportunity score that prioritizes refresh candidates.

**Action**

Review and update the highest-priority pages by improving content quality, metadata, internal links, or SEO strategy.

**Cost of a wrong recommendation**

If the recommendations are inaccurate, high-value pages may be overlooked while time is spent refreshing pages with little improvement potential. This reduces the effectiveness of content optimization efforts.

**Why machine learning?**

Search performance depends on multiple interacting signals such as impressions, clicks, CTR, engagement, freshness, and trends. Machine learning can combine these signals to prioritize refresh opportunities more effectively than simple rule-based approaches.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [5]:
!git clone https://github.com/joj48/Flyrank.ai-SEO-Project.git
import os
os.chdir('Flyrank.ai-SEO-Project')
import pandas as pd
df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

print(f"Rows: {df.shape[0]}")
print(f"Columns: {df.shape[1]}")
print(f"Unique clients: {df['client_id'].nunique()}")

print(df[['impressions_90d',
          'clicks_90d',
          'ctr',
          'engagement_rate']].describe())

Cloning into 'Flyrank.ai-SEO-Project'...
remote: Enumerating objects: 159, done.
remote: Counting objects: 100% (159/159), done.
remote: Compressing objects: 100% (111/111), done.
remote: Total 159 (delta 65), reused 104 (delta 32), pack-reused 0 (from 0)
Receiving objects: 100% (159/159), 1.86 MiB | 14.96 MiB/s, done.
Resolving deltas: 100% (65/65), done.
Rows: 30000
Columns: 44
Unique clients: 32
       impressions_90d    clicks_90d           ctr  engagement_rate
count     30000.000000  30000.000000  30000.000000     30000.000000
mean       5200.366300     16.097333      0.510733         2.534520
std       16838.019547     75.076958      3.279162         8.310096
min           1.000000      0.000000      0.000000         0.000000
25%          81.000000      0.000000      0.000000         0.000000
50%         731.000000      1.000000      0.070000         0.000000
75%        3615.250000      7.000000      0.290000         1.350000
max      517715.000000   4178.000000    100.000000    

### Observations from the Starter Dataset

The starter dataset contains **30,000 content items** across **32 anonymized clients**, providing a sufficiently large dataset for content opportunity analysis.

The average number of impressions over the last 90 days is **5,200.37**, while the median is only **731**. This large difference suggests that content performance varies significantly across pages, making prioritization an important problem.

The average CTR is **0.51%**, and many pages have very low click performance, indicating potential opportunities to identify content that may benefit from refresh or optimization.

These observations support the **Refresh / Content Opportunity Scoring** lane because the dataset contains meaningful variation in search performance and engagement signals that can be used to prioritize content review.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

This work is intended to provide decision support for prioritizing webpages that should be reviewed for content refresh. The recommendations are based on observed patterns in the available dataset and should not be interpreted as proof of causation or guaranteed future search performance. The target used in the starter dataset is a proxy based on historical trends, so the results should be viewed as a ranking of opportunities rather than definitive predictions.

## Self-check

Before you submit, confirm each line honestly:

- [x] Every section above is filled — markdown thinking AND the code that backs it
- [x] The notebook runs top to bottom with no errors (Runtime → Run all)
- [x] No client names, URLs, or private queries anywhere
- [x] My claims use careful words: observed, measured, directional, decision-support
- [x] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.